velox/connectors/hive/HiveConfig.cpp

```diff
@@ -272,6 +272,16 @@ bool HiveConfig::s3UseProxyFromEnv() const {
   return config_->get<bool>(kS3UseProxyFromEnv, false);
 }
 
+uint8_t HiveConfig::readTimestampUnit(const Config* session) const {
```

Comment:
The unit should be read from the Parquet logical type for this column, not set by the user as a config property. See https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#timestamp

Reply:
This is corresponding to the timestamp unit that can be handled in compute engine, not related to the type in the file.  It should be the more coarse one of both.

Reply:
@yingsu00 In a Parquet file, the unit of int96 is fixed because it is made up of days and nanos, unlike int64-timestamp, which can have different units. We parse days and nanos from Parquet, while the compute engine may need different units of timestamps, e.g. Presto needs milli while Spark needs micro. This config allows us to adjust timestamp precision according to user's requirement.

Without this change, the filter result could become incorrect. For example, for a Spark filter `a == 2000-09-12 22:36:29.000000`, if a is stored as nano unit in Velox, when a is `2000-09-12 22:36:29.000000111` Velox returns false but Spark needs true because it only cares about the micro digits.

Therefore, we need to truncate the value and this logic is also needed for int64-timestamp reader. Does this makes sense? Thanks.

Reference for Int96 in Parquet: https://github.com/apache/parquet-format/pull/49/files#diff-0e877db0daf579f98a11e5e113b29250a2dcae3decb1e83a88db1e6f092bee96R149-R150

Reply:
Thank you for helping review this PR.